main.cpp

```diff
@@ -0,0 +1,87 @@
+#include <Arduino.h>
+#include <iostream>
+
+#define temp_PIN 4 // pin numbers
+#define brakePressure_PIN 5
+#define coolantPressure_PIN 6
+
+#define wheelSpeed_PIN 14
+
+#define tempConst 5 // adjustable temperature proportionality constant for testing
+
+uint16_t wheelPulses=120; // set at random number for testing

```

Comment:
this could be a potential and maybe the potential issue! When dealing with variables and interrupts, you should declare variables as volatile.

The compiler may try to be a smart aleck and see that the ISR function never gets explicitly called. It may just assume wheelPulses never gets incremented and so will lazily never update the value of wheelPulses properly in the main code (even if wheelPulses is actually being incremented!). 

The ISR depends on hardware signals that the compiler wouldn't be aware of. By making it volatile, you're basically saying in plain english "I know more about what will happen to this variable than you. **Please don't do any smart optimizations**. Read the variable from memory every single time"